composer/models/huggingface.py

```diff
@@ -274,9 +274,9 @@ def loss(self, outputs, batch):
             return outputs[0]
 
     def eval_forward(self, batch, outputs: Optional[Any] = None):
+        self.labels = batch.pop('labels')
```

Comment:
yeah basically, if the labels are part of the batch when you call forward, then the HF model will calculate cross entropy loss, which is a waste of time for the ICL eval metrics

Reply:
Oh, I see. I think this might break some other functionality though, because other uses might assume that the output will contain the loss if `self.use_logits` is `False`. do you think it would make sense to only pop the labels before calling `forward` `if self.use_logits or batch.get('mode', None) == 'icl_task':`?